osquery/tables/system/windows/disk_info.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genDiskInfo(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+  WmiRequest wmiSystemReq("select * from Win32_DiskDrive");
+  std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  if (!wmiResults.empty()) {
```

Comment:
You could make this a bit tighter if you flipped the logic here and just use an informational message as opposed to passing back `-1`. So maybe something like:
```
if (wmiResults.empty()) {
  LOG(INFO) << "WMI Query for Disk Information failed";
  return results;
}
for (const auto& data : wmiResults) {
  long partitionCount = 0;
  long diskIndex = 0;
  data.GetLong("Partitions", partitionCount);
  r["partitions"] = INTEGER(partitionCount);
  data.GetLong("Index", diskIndex);
  r["disk_index"] = INTEGER(diskIndex);
  data.GetString("InterfaceType", r["type"]);
  data.GetString("PNPDeviceID", r["pnp_device_id"]);
  data.GetString("DeviceID", r["device_id"]);
  data.GetString("Size", r["disk_size"]);
  data.GetString("Manufacturer", r["manufacturer"]);
  data.GetString("Model", r["hardware_model"]);
  data.GetString("Name", r["name"]);
  data.GetString("SerialNumber", r["serial"]);
  data.GetString("Description", r["description"]);
  results.push_back(r);
}
```

Reply:
That could do, though I do have a question regarding that. Since most other Windows tables return -1 on all values if there is an error, would it be better to keep it for consistency between cpp files? If that's not a concern, then I can change it to write out a message instead.

Reply:
I'd say with this a warning message would be more meaningful over a series of -1 columns. In fact you might even considering making this a `LOG(WARNING)` before you bail out, as having a table with nothing but `-1` doesn't give much information about what went wrong.

Reply:
How about this? I kept the if, else statement the same, but replaced the else logic with a warning message saying that the resultset is empty.

Reply:
I guess I'm not sure what you're concern is with pulling the `for` logic out of the `if`/`else`? Doing it that way will reduce a level of scope in your code and makes it a little cleaner. I'd still prefer it if you just bail early on failure.